components/nodemanager-service/pgdb/project_delete.go

```diff
@@ -0,0 +1,89 @@
+package pgdb
+
+import (
+	"context"
+
+	project_update_lib "github.com/chef/automate/lib/authz"
+	uuid "github.com/chef/automate/lib/uuid4"
+	log "github.com/sirupsen/logrus"
+)
+
+const deleteProjectNodeAssociation = `
+DELETE FROM nodes_projects np
+USING projects p
+WHERE p.id = np.project_id AND p.project_id = $1;
+`
+
+type ProjectDelete struct {
+	jobStatus      project_update_lib.JobStatus
+	jobStatusError error
+	running        bool
+	ID             string
+}
+
+func (db *DB) DeleteProjectTag(ctx context.Context,
+	projectTagToBeDelete string) ([]string, error) {
+	jobUUID := uuid.Must(uuid.NewV4()).String()
+	log.Debugf("Running node manager DeleteProjectTag %v ID: %s", projectTagToBeDelete, jobUUID)
+
+	db.ProjectDelete.start(jobUUID)
+
+	go db.deleteProject(projectTagToBeDelete)
+
+	return []string{jobUUID}, nil
+}
+
+func (db *DB) deleteProject(projectTagToBeDelete string) {
+	_, err := db.Exec(deleteProjectNodeAssociation, projectTagToBeDelete)
+	if err != nil {
+		db.ProjectDelete.updateFail(err)
+	}
+
+	db.ProjectDelete.updateComplete()
+}
+
+func (projectDelete *ProjectDelete) start(jobUUID string) {
+	projectDelete.jobStatusError = nil
+	projectDelete.running = true
+	projectDelete.jobStatus = project_update_lib.JobStatus{
+		Completed:             false,
+		PercentageComplete:    0,
+		EstimatedEndTimeInSec: 0,
+	}
+	projectDelete.ID = jobUUID
+}
+
+func (projectDelete *ProjectDelete) getJobStatus(jobUUID string) (project_update_lib.JobStatus, error) {
+	if projectDelete.ID == jobUUID {
+		if projectDelete.jobStatusError != nil {
+			return project_update_lib.JobStatus{}, projectDelete.jobStatusError
+		}
+		return projectDelete.jobStatus, nil
+	}
+
+	// If the jobID does not match the currently running job then it must have completed before.
+	return project_update_lib.JobStatus{
+		Completed:             true,
+		PercentageComplete:    1,
+		EstimatedEndTimeInSec: 0,
+	}, nil
+}
+
+func (projectDelete *ProjectDelete) updateComplete() {
+	projectDelete.jobStatus = project_update_lib.JobStatus{
+		Completed:             true,
+		PercentageComplete:    1.0,
```

Comment:
Why is this `PercentageComplete` a float while the other is a bool?

Reply:
`PercentageComplete` is the range between 1.0 to 0.0 so it can not be an integer unless we change the range to 100 to 0. The `EstimatedEndTimeInSec` is the number of seconds since 1970.

Reply:
It appears to be an integer here: https://github.com/chef/automate/pull/2424/files#diff-87e11a1c44a032343e1ce6928c55b7c7R67 and here https://github.com/chef/automate/pull/2424/files#diff-35fc59428b9b5a554bcf8b413be709ddR131

Maybe converted?

Reply:
The underlying data is stored as a float defined here https://github.com/chef/automate/blob/b5f2710d53bbad79a7a4602bd0799f19d40407fb/lib/authz/project_rules.go#L9. 

So, it is the same if we have a `1.0` or a `1` it is stored the same.